### ISSUE TITLE
build fails at compile-picard

### ISSUE THREAD

**Issue by sgivan** [broadinstitute/picard#352]
In my hands, I can't get picard to build. I've followed the directions in the README.md on github and also tried the directions in the documentation about setting JAVA6_HOME and/or setting the -Djava6.home option with ant. Both seem to fail with the following message. I've tried building with either Java 7 or Java 8.

```

compile-picard:
    [javac] Compiling 244 source files to /share/apps/temp/picard/classes
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 1.6
    [javac] /share/apps/temp/picard/src/java/picard/illumina/parser/readers/BclQualityEvaluationStrategy.java:35: warning: '_' used as an identifier
    [javac]                 public AtomicInteger make(final Byte _) {
    [javac]                                                      ^
    [javac]   (use of '_' as an identifier might not be supported in releases after Java SE 8)
    [javac] /share/apps/temp/picard/src/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java:56: error: incompatible types: List<MetricBase> cannot be converted to List<PreAdapterDetailMetrics>
    [javac]         final List<PreAdapterDetailMetrics> preAdapterDetailMetricsList = (List<PreAdapterDetailMetrics>) MetricsFile.readBeans(PRE_ADAPTER_IN);
    [javac]                                                                                                                                ^
    [javac] /share/apps/temp/picard/src/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java:57: error: incompatible types: List<MetricBase> cannot be converted to List<BaitBiasDetailMetrics>
    [javac]         final List<BaitBiasDetailMetrics> baitBiasDetailMetricsList = (List<BaitBiasDetailMetrics>) MetricsFile.readBeans(BAIT_BIAS_IN);
    [javac]                                                                                                                          ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 2 errors
    [javac] 2 warnings

BUILD FAILED
/share/apps/temp/picard/build.xml:171: The following error occurred while executing this line:
/share/apps/temp/picard/build.xml:535: Compile failed; see the compiler error output for details.

```

**Comment by tfenne** [comment]
Hi @sgivan - yes the build is currently broken because of a change in HTSJDK that needs a compensating change in Picard.  That change (eliminating those two cases ) is in pR #350, which is tied up because the travis builds are failing because HTSJDK now requires Java 8, and Picard isn't setup to build with Java 8.  And ... the Picard tests fail on Java 8 due to some poorly written tests that are reliant on the ordering produced by some HashSets/HashMaps and that behaviour changed between Java 7 and Java 8.

Long story short, remove the casts from those two lines and you'll get a working build until we get this mess straightened out.

**Comment by sgivan** [comment]
OK, thanks. I'm sort of glad to hear it's not something I'm doing wrong.

**Comment by lbergelson** [comment]
You can also cd into the htsjdk directory and 'git checkout 1.141' which should set the htsjdk version to the last good one.

**Comment by sgivan** [comment]
Thanks @lbergelson

**Comment by vdauwera** [comment]
This has since been resolved.
